syntax = "proto3";

package endpoints;

// This message is included for potential future use.
message GetArtifactConfig {
}

message BatchGetArtifactRequest {
  map<string, GetArtifactConfig> configs_by_artifact_id = 1;
}

message GetArtifactResult {
  // Required field indicating the number of parts uploaded by the artifact.
  optional uint32 part_count = 1;

  // Required field indicating the precise number of bytes each part contains,
  // other than the final part, which can contain fewer bytes.
  optional uint64 part_size_threshold = 2;

  // Required field indicating the size in bytes of the combined parts.
  optional uint64 total_size = 3;
}

message BatchGetArtifactResponse {
  map<string, GetArtifactResult> results_by_artifact_id = 1;
}
